### PR TITLE
Native quantum control toml fixup

### PR DESCRIPTION
**Pull request by grwlf**
**Context:** Ongoing transition to the schema 2 of the quantum device toml specification

**Description of the Change:** This is a minor source code clean up PR.
* Remove force addition of the ControlQubitUnitary gate to the set of QJITDevice operations. This gate is handled by the generic logic in schema 2 config loader. In schema 1 loader, the sources already have all the required patches.
* Explicitly list the runtime-supported controlled gates
* Fix missing Projector observable in the lightning-kokkos device config
* Open toml files in binary mode (fixes wheel builds)
* Update some comments and simplify the set calculations.

**Benefits:** Cleaner sources

**Possible Drawbacks:** N/A

**Related GitHub Issues:** https://github.com/PennyLaneAI/catalyst/pull/554

### CODE REVIEW

File: frontend/catalyst/qjit_device.py
Comment (dime10) [inline frontend/catalyst/qjit_device.py:85]: These are not runtime operations (there is no gate named "C(Hadamard)" in the runtime interface), and I definitely don't want us embracing this format rather than moving away from it.

Comment (grwlf) [reply frontend/catalyst/qjit_device.py:85]: In this PR I just replaced the runtime calculation of the same set with a hardcoded names. This allows me to remove [this call](https://github.com/PennyLaneAI/catalyst/pull/600/files#diff-49c4bb20bf2dcbd5fe316c548577fc168c4da1b075304b5bb953afcf8c4c7c25L77) which did the same.
I don't argue to keep these in the long term. As I see it, we remove the whole list when the new-style decomposition is ready

Comment (grwlf) [reply frontend/catalyst/qjit_device.py:85]: @dime10 but I can remove this change if you really don't want to touch this. Let me know.

**Comment by dime10** [comment]
@grwlf Should we merge this after the lightning update is merged so that the TODOs can be removed?

**Comment by grwlf** [comment]
> @grwlf Should we merge this after the lightning update is merged so that the TODOs can be removed?

It is possible but I don't think it is necessary. It might be easier to remove TODOs by a separate PR, depending on how fast they approve the PR in lightning.

File: .github/workflows/check-catalyst.yaml
Comment (dime10) [inline .github/workflows/check-catalyst.yaml:389]: This should be reverted I think since it was handled in a separate PR

**Review by dime10 — APPROVED** [review]

File: .github/workflows/check-catalyst.yaml
Comment (grwlf) [reply .github/workflows/check-catalyst.yaml]: Done

**Pull request merged**
